storage/src/main/java/com/tencent/rss/storage/handler/impl/HdfsClientReadHandler.java

```diff
@@ -51,7 +66,96 @@ public HdfsClientReadHandler(
         ShuffleStorageUtils.getShuffleDataPathWithRange(appId,
             shuffleId, partitionId, partitionNumPerRange, partitionNum));
     init(fullShufflePath);
-    readAllIndexSegments();
+    cursor = indexReaderMap
+        .entrySet()
+        .stream()
+        .sorted(Comparator.comparing(Entry::getKey))
+        .collect(Collectors.toList())
+        .iterator();
+  }
+
+  public ShuffleIndexResult readShuffleIndex() {
+    if (!cursor.hasNext()) {
+      return null;
+    }
+
+    Entry<String, HdfsFileReader> entry = cursor.next();
+    cursorPath = entry.getKey();
+    long start = System.currentTimeMillis();
+    HdfsFileReader reader = entry.getValue();
+    if (reader == null) {
+      LOG.error("Reader of {} is invalid!", cursorPath);
+      return null;
```

Comment:
Should we return null or empty ShuffleIndexResult?  Can we unify these?

Reply:
Ok, this would make readShuffleData remove the null conditiion check.